src/Utils/RMC/RMST.as

```diff
@@ -246,22 +236,23 @@ class RMTS : RMS {
                 if (currentMapChallenge !is null) {
                     CGameCtnChallengeInfo@ currentMapInfo = currentMapChallenge.MapInfo;
                     if (currentMapInfo !is null) {
-                        RMC::TimePlayed = Time::Now - RMC::RunStartTimestamp - RMC::TimePaused;
-                        RMC::TimeSpentMap = Time::Now - RMC::SpawnedMapTimestamp;
-                        if (!RMC::IsRunning || RunRemainingTime() < 1) {
+                        RMC::StartTime = Time::Now;
+                        RMC::TimeSpentMap = Time::Now - RMC::TimeSpawnedMap;
+                        PendingTimerLoop();
+                        if (!RMC::IsRunning || (RMC::EndTime - Time::Now) < 1) {
+                            RMC::StartTime = -1;
+                            RMC::EndTime = -1;
                             RMC::IsRunning = false;
-                            RMC::ContinueSavedRun = false;
+                            RMC::ShowTimer = false;
                             GameEndNotification();
-
                             @nextMap = null;
                             @MX::preloadedMap = null;
                             m_playerScores.SortDesc();
-
-                            BetterChatSendMessage(
-                                Icons::Users + " Random Map Together Survival ended.\n"
-                                "Thanks for playing!"
-                            );
+#if DEPENDENCY_BETTERCHAT
+                            BetterChat::SendChatMessage(Icons::Scuttlebutt + " RMS Together ended, thanks for playing!");
+                            sleep(200);
```

Comment:
I think `BetterChatSendMessage` wraps this preprocessor condition and `sleep`, unless I'm mistaken (same with `BetterChatSendLeaderboard`?).

Reply:
Order of things seems to be important, which is spooky